src/kube.documentProvider.ts

```diff
@@ -0,0 +1,35 @@
+import * as vscode from "vscode";
+import * as querystring from "querystring";
+import { Kubectl } from "./kubectl";
+
+/// a kubernetes content provider provides json content for kubernetes resources
+export class KubernetesDocumentProvider implements vscode.TextDocumentContentProvider {
+    public constructor(private kubectl: Kubectl) {
+
+    }
+    provideTextDocumentContent(uri: vscode.Uri, token: vscode.CancellationToken): vscode.ProviderResult<string> {
+        return new Promise<string>(async (resolve, reject) => {
```

Comment:
hmm - perhaps pull this anonymous function out into its own named async function on this class?

Reply:
provideTextDocumentContent(uri: vscode.Uri, token: vscode.CancellationToken): vscode.ProviderResult<string> will compile error if you add async because ProviderResult is not compatible with Promise

Reply:
async provideTextDocumentContent(uri: vscode.Uri, token: vscode.CancellationToken): Promise<string>  seems work.

Reply:
Changed accordingly